samples/drivers/adc/adc_dt/src/main.c

```diff
@@ -54,7 +54,11 @@ int main(void)
 		}
 	}
 
+#ifndef CONFIG_COVERAGE
 	while (1) {
+#else
+	for (int k = 0; k < 10; k++) {
```

Comment:
10 is a magic number

Reply:
Indeed, but very similar to other changed eg. #71207
That number has no meaning, there are just needed a few loops.

Reply:
exactly as @nordic-piks explained, no of loops has to be finite and low enough to not exceed test timeout (together with time needed for dumping coverage data). 
Eventually we can even drop #else section and have one execution - that's enough, but I think it's nothing wrong to have few readings in output.